### PR TITLE
Adding CredentialStatus to data model

### PR DESCRIPTION
**Pull request by simonas-notcat**
Fixes: #129 and adds CredentialStatus to the data model

### CODE REVIEW

File: packages/daf-cli/src/credential.ts [hunk 1]
```diff
@@ -41,22 +42,55 @@ program
         message: 'Claim Value',
         default: 'Alice',
       },
+      {
+        type: 'list',
+        name: 'addStatus',
+        message: 'Add credential status info?',
```
Comment (mirceanis) [inline packages/daf-cli/src/credential.ts:48]:
```suggestion
        message: 'Is the credential revocable?',
```

File: packages/daf-w3c/src/action-handler.ts [hunk 2]
```diff
@@ -109,54 +114,80 @@ export interface PresentationInput {
...
+        result['vc']['credentialSubject'] = credentialSubject
+      break
+      case '@context':
+      case 'context':
+        result['vc']['@context'] = input[key]
+      break
+      case 'type':
+        result['vc']['type'] = input[key]
+      break
+      case 'expirationDate':
```
Comment (mirceanis) [inline packages/daf-w3c/src/action-handler.ts:139]: There should also be a case for  `issuanceDate` that translates to `nbf`
Same transformations about date apply to presentations too and should also be done in reverse when parsing JWT

File: packages/daf-w3c/src/action-handler.ts [hunk 3]
```diff
@@ -109,54 +114,80 @@ export interface PresentationInput {
   tag?: string
   expirationDate?: string
   verifiableCredential: string[]
+  [x: string]: any
 }
 
 const transformCredentialInput = (input: CredentialInput): VerifiableCredentialPayload => {
```
Comment (awoie) [inline packages/daf-w3c/src/action-handler.ts:120]: Are we dropping custom claims outside of the `credentialSubject`, e.g., 

```
{
  "@context": [
    "https://www.w3.org/2018/credentials/v1",
    "https://www.w3.org/2018/credentials/examples/v1"
  ],
  "id": "http://example.edu/credentials/1872",
  "type": ["VerifiableCredential", "AlumniCredential"],
  "issuer": "https://example.edu/issuers/565049",
  "issuanceDate": "2010-01-01T19:73:24Z",
  "customProperty": "12345",
  "credentialSubject": {
    "id": "did:example:ebfeb1f712ebc6f1c276e12ec21",
    "alumniOf": {
      "id": "did:example:c276e12ec21ebfeb1f712ebc6f1",
    }
  }
```

Comment (awoie) [inline packages/daf-w3c/src/action-handler.ts:120]: What happens if `credentialSubject` is an array? I don't think we should support this at the moment but the spec supports it, so we should handle that case with an error.

Comment (mirceanis) [reply packages/daf-w3c/src/action-handler.ts:120]: we were before, but no longer after this PR

**Pull request merged**